lib/Settings.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
+var nconf = require('nconf');
+var Promise = require('bluebird');
+
+var defined = Cesium.defined;
+
+var Settings = {};
+/** Check for necessary settings in `configPath` file and initialize `nconf`
+ *
+ * @param {String} configPath Path to configuration file (usually `./config.json`)
+ * @return {Promise<String[] | String>} Repository names or Error message if Promise fails
+ */
+Settings.loadRepositoriesSettings = function (configPath) {
+    nconf.env('__')
+        .file({
+            file: configPath
+        })
+        .defaults({
+            port: 5000,
+            listenPath: '/'
+        })
+        .required([
+            'secret',
+            'repositories'
+        ]);
+
+    return fsExtra.readJson(configPath)
```

Comment:
Why do we need to read this file a second time?  Shouldn't all of the data be loaded in nconf already?